Run-SystemTest.ps1

```diff
@@ -51,19 +55,50 @@ Process {
     try {
         Push-Location $workingDir
 
+
         # Build the OpenApi.MSBuild package to be used in the system tests
         Exec { & dotnet pack -c Release -o "$outputDir" }
 
+        ### Testing Generate Contract Mode ###
+
+        # Then Should Successfully Build
+        BuildProject -openApiMsBuildSource $outputDir -projectPath $genericSysTestDir -isFailureExpected $false -extraArgs "/p:OpenApiDevelopmentMode=GenerateContract"
+        # When using legacy name / Then Should Successfully Build
         BuildProject -openApiMsBuildSource $outputDir -projectPath $genericSysTestDir -isFailureExpected $false -extraArgs "/p:OpenApiDevelopmentMode=CodeFirst"
+        # When Comparing Spec and No Diff Error / Then Should Successfully Build 
+        BuildProject -openApiMsBuildSource $outputDir -projectPath $genericSysTestDir -isFailureExpected $false -extraArgs "/p:OpenApiDevelopmentMode=GenerateContract;OpenApiCompareCodeAgainstSpecFile=true"
+        # When Comparing Spec and Have Diff / Then Should Fail Build
+        BuildProject -openApiMsBuildSource $outputDir -projectPath $oasDiffErrorSysTestDir -isFailureExpected $true -extraArgs "/p:OpenApiDevelopmentMode=GenerateContract;OpenApiCompareCodeAgainstSpecFile=true"
+
+
+        ### Testing Compare Contract Mode ###
+
+        # Given no diff / Then Should Successfully Build
+        BuildProject -openApiMsBuildSource $outputDir -projectPath $genericSysTestDir -isFailureExpected $false -extraArgs "/p:OpenApiDevelopmentMode=ValidateContract"
```

Comment:
I don't think we need this now but it would make it safer to have more isolate test, potential risk are:

- If somebody run OpenAPI in Generate Contract mode on the OasDiffError project, there won't be an error anymore

Cleaning could be tricky here since we can't simply delete a file since it won't be the initial state and to revert changes could be tricky on the CI. We would probably need to copy the whole project for each test instance and delete the copy after but this seems out-of-scope.